exo-go/src/app_dependency_helpers/index_test.go

```diff
@@ -0,0 +1,167 @@
+package appDependencyHelpers_test
+
+import (
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/Originate/exosphere/exo-go/src/app_config_helpers"
+	"github.com/Originate/exosphere/exo-go/src/app_dependency_helpers"
+	"github.com/Originate/exosphere/exo-go/src/types"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("AppDependency", func() {
+	var appConfig types.AppConfig
+	var appDir string
+	var exocom appDependencyHelpers.AppDependency
+	var nats appDependencyHelpers.AppDependency
+	var mongo appDependencyHelpers.AppDependency
+
+	var _ = BeforeSuite(func() {
+		appDir = path.Join("..", "..", "..", "exosphere-shared", "example-apps", "complex-setup-app")
+		var err error
+		appConfig, err = appConfigHelpers.GetAppConfig(appDir)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	var _ = Describe("Build", func() {
+		It("should build each dependency successfully", func() {
+			for _, dependency := range appConfig.Dependencies {
+				builtDependency := appDependencyHelpers.Build(dependency, appConfig, appDir)
+				switch dependency.Name {
+				case "exocom":
+					exocom = builtDependency
+				case "mongo":
+					mongo = builtDependency
+				}
+			}
+			nats = appDependencyHelpers.Build(types.Dependency{
+				Name:    "nats",
+				Version: "0.9.6",
+			}, appConfig, appDir)
```

Comment:
This appears to be setting `nats` for all future tests in this file. The initialization of nats should be done in a beforeEach block. We want all the `it` to not effect each other and can use beforeEach for common setup needed between tests. Each test should start with a clean slate. Thoughts on restructuring these tests to be the following:
```go
var _ = Describe("AppDependency", func() {
  var _ = Describe("Build", func() {})
  
  var _ = Describe("nats dependency", func() { 
    // Tests for GetContainerName / GetDokerConfig / etc 
    // a beforeEach creates the nats variable used in each test
  })
  
  var _ = Describe("exocom dependency", func() {
    // similar to nats dependency
  })
  
  var _ = Describe("generic dependency", func() {
    // similar to nats dependency
  })
})
```